ShareClientUI/ShareClientSettingsViewController.swift

```diff
@@ -15,16 +16,30 @@ public class ShareClientSettingsViewController: UITableViewController {
 
     public let cgmManager: ShareClientManager
 
-    private var glucoseUnit: HKUnit
+    private let displayGlucoseUnitObservable: DisplayGlucoseUnitObservable
+
+    private var cancellable: AnyCancellable?
+
+    private var glucoseUnit: HKUnit {
+        displayGlucoseUnitObservable.displayGlucoseUnit
+    }
 
     public let allowsDeletion: Bool
 
-    public init(cgmManager: ShareClientManager, glucoseUnit: HKUnit, allowsDeletion: Bool) {
+    public init(cgmManager: ShareClientManager, displayGlucoseUnitObservable: DisplayGlucoseUnitObservable, allowsDeletion: Bool) {
         self.cgmManager = cgmManager
-        self.glucoseUnit = glucoseUnit
+        self.displayGlucoseUnitObservable = displayGlucoseUnitObservable
         self.allowsDeletion = allowsDeletion
 
         super.init(style: .grouped)
+
+        cancellable = displayGlucoseUnitObservable.updatePublisher.sink { [weak self] in
```

Comment:
[nit] Consider above comment, plus:
```
$displayGlucoseUnitObservable
    .sink { [weak self] in self?.tableView.reloadData() }
    .store(in: &cancellables)
```
Not sure if this will work here.